core/src/main/java/com/scalar/db/common/CommonDistributedStorageAdmin.java

```diff
@@ -363,6 +363,50 @@ public void renameColumn(
     }
   }
 
+  @Override
+  public void alterColumnType(
+      String namespace, String table, String columnName, DataType newColumnType)
+      throws ExecutionException {
+    TableMetadata tableMetadata = getTableMetadata(namespace, table);
+    if (tableMetadata == null) {
+      throw new IllegalArgumentException(
+          CoreError.TABLE_NOT_FOUND.buildMessage(ScalarDbUtils.getFullTableName(namespace, table)));
+    }
+
+    if (!tableMetadata.getColumnNames().contains(columnName)) {
+      throw new IllegalArgumentException(
+          CoreError.COLUMN_NOT_FOUND2.buildMessage(
+              ScalarDbUtils.getFullTableName(namespace, table), columnName));
+    }
+
+    if (tableMetadata.getPartitionKeyNames().contains(columnName)
+        || tableMetadata.getClusteringKeyNames().contains(columnName)
+        || tableMetadata.getSecondaryIndexNames().contains(columnName)) {
+      throw new IllegalArgumentException(
+          CoreError.ALTER_PRIMARY_OR_INDEX_KEY_COLUMN_TYPE_NOT_SUPPORTED.buildMessage(
+              ScalarDbUtils.getFullTableName(namespace, table), columnName));
+    }
+
+    DataType currentColumnType = tableMetadata.getColumnDataType(columnName);
+    if (currentColumnType == newColumnType) {
+      return;
+    }
+    if (!isTypeConversionSupported(currentColumnType, newColumnType)) {
+      throw new IllegalArgumentException(
+          CoreError.INVALID_COLUMN_TYPE_CONVERSION.buildMessage(
```

Comment:
I see there is a little inconsistency between the error message ("Invalid column type conversion from ...") and the method name (`isTypeConversionSupported`).

There is a `JDBC_UNSUPPORTED_COLUMN_TYPE_CONVERSION`, so should they be all merged (like by defining `UNSUPPORTED_COLUMN_TYPE_CONVERSION` or is there a particular reason for separating them?

Reply:
I'm fine with separating them if there is a good reason, but I would like the error message and the method a bit more consistent.
How about naming the method `isTypeConversionValid` or something like that?